engine/src/main/scala/cromwell/engine/workflow/workflowstore/package.scala

```diff
@@ -0,0 +1,59 @@
+package cromwell.engine.workflow
+
+import cromwell.core.{WorkflowId, WorkflowSourceFiles}
+
+import scala.concurrent.{ExecutionContext, Future}
+import scalaz.NonEmptyList
+
+package object workflowstore {
+  sealed trait WorkflowState {
+    def isStartable: Boolean = {
+      this match {
+        case x: StartableState => true
+        case _ => false
+      }
+    }
+  }
+
+  case object Running extends WorkflowState
+
+  sealed trait StartableState extends WorkflowState
+  case object Submitted extends StartableState
+  case object Restartable extends StartableState
+
+  final case class SubmittedWorkflow(id: WorkflowId, sources: WorkflowSourceFiles, state: WorkflowState) {
+    def toWorkflowToStart: WorkflowToStart = {
+      state match {
+        case r: StartableState => WorkflowToStart(id, sources, r)
+        case _ => throw new IllegalArgumentException("This workflow is not currently in a startable state")
+      }
+    }
+  }
+
+  final case class WorkflowToStart(id: WorkflowId, sources: WorkflowSourceFiles, state: StartableState)
+
+  /**
+    * Stores any submitted workflows which have not completed.
+    *
+    * This is not thread-safe, although it should not be being used in a multithreaded fashion. If that
+    * changes, revisit.
+    */
+  trait WorkflowStoreDatabase {
```

Comment:
I agreed that the workflow state definitions shouldn't belong to the Database trait. But the trait itself is very useful as an interface